hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java

```diff
@@ -273,24 +275,24 @@ private TServer newThriftServer(TServerSocket socket, int poolSize, HiveConf con
   }
 
   private void initConf(HiveConf conf, int port, boolean directSql) {
-    conf.set(HiveConf.ConfVars.METASTOREURIS.varname, "thrift://localhost:" + port);
-    conf.set(
-        HiveConf.ConfVars.METASTOREWAREHOUSE.varname, "file:" + HIVE_LOCAL_DIR.getAbsolutePath());
-    conf.set(HiveConf.ConfVars.METASTORE_TRY_DIRECT_SQL.varname, String.valueOf(directSql));
-    conf.set(HiveConf.ConfVars.METASTORE_DISALLOW_INCOMPATIBLE_COL_TYPE_CHANGES.varname, "false");
+    conf.set("hive.metastore.uris", "thrift://localhost:" + port);
+    conf.set("hive.metastore.warehouse.dir", "file:" + HIVE_LOCAL_DIR.getAbsolutePath());
+    conf.set("hive.metastore.try.direct.sql", String.valueOf(directSql));
+    conf.set("hive.metastore.disallow.incompatible.col.type.changes", "false");
     conf.set("iceberg.hive.client-pool-size", "2");
     // Setting this to avoid thrift exception during running Iceberg tests outside Iceberg.
-    conf.set(
-        HiveConf.ConfVars.HIVE_IN_TEST.varname, HiveConf.ConfVars.HIVE_IN_TEST.getDefaultValue());
+    conf.set("hive.in.test", "false");
   }
 
   private static void setupMetastoreDB(String dbURL) throws SQLException, IOException {
     try (Connection connection = DriverManager.getConnection(dbURL)) {
       ScriptRunner scriptRunner = new ScriptRunner(connection, true, true);
+      String resource =
+          (HiveVersion.current() == HiveVersion.HIVE_4)
+              ? "hive-schema-4.0.0.derby.sql"
+              : "hive-schema-3.1.0.derby.sql";
```

Comment:
nit: Maybe a comment here that Hive 2.3.x and 3.1.x can use the same HMS DB backend schema?

Not strong requirement, just took some time for me to understand

Reply:
Yes, I'll add a comment. I think I pointed this out in a comment on https://github.com/apache/iceberg/pull/12681 (but not in the code).

Reply:
Done.